### PR TITLE
Changes the way slayer distributes weapons.

### PR DESCRIPTION
**Pull request by X0-11**
:cl: XO-11
tweak: Slayer now distributes weapons around the map differently and consistently respawns them allowing for longer and more interesting slayer matches.
tweak: Air vehicles no longer just fall to the ground from a single EMP hit.
tweak: Vehicle smokescreen dispersal has been tweaked.
/:cl:

### CODE REVIEW

File: code/modules/halo/vehicles/vehiclebase.dm [hunk 1]
```diff
@@ -713,12 +713,13 @@
 		if(new_time > spawn_datum.emp_toggle_time)
 			spawn_datum.emp_toggle_time = new_time
 
+	/*
```
Comment (bloxgate) [inline code/modules/halo/vehicles/vehiclebase.dm]: This should probably get a changelog entry too

File: code/modules/halo/weapons/forerunner/snipers.dm [hunk 2]
```diff
@@ -15,7 +15,7 @@
 	reload_sound = 'code/modules/halo/sounds/binaryrifle_reload.ogg'
 	one_hand_penalty = -1
 	scoped_accuracy = 7
-	accuracy = -10
+	accuracy = -9
```
Comment (bloxgate) [inline code/modules/halo/weapons/forerunner/snipers.dm:18]: Might also want to consider a CL entry for this as well

File: maps/slayer/slayer_itemspawn_markers.dm [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	icon = 'icons/mob/screen1.dmi'
+	icon_state = "x4"
+	invisibility = 101
+	var/spawn_interval = 2 MINUTES
+	var/time_next_item_spawn = 0
+	var/obj/last_spawned_item //We track our last spawned item, until we realise it's moved off of our position.
+	var/list/spawnables = list()
+	var/mags_spawn = 0
+
+/obj/effect/itemspawn_marker/process()
```
Comment (bloxgate) [inline maps/slayer/slayer_itemspawn_markers.dm]: Just want to double check, these are added to processing_objects and stay there? I don't seem to see a default process proc for `/obj/effect` and the one for `/obj` will remove itself from processing_objects when called.

I also don't see any code for the default effect that actually puts it in processing_objects, but I may just be searching for the wrong term.

**Review by bloxgate — CHANGES_REQUESTED** [review]

File: maps/slayer/slayer_itemspawn_markers.dm (hunk 3 above)
Comment (X0-11) [reply maps/slayer/slayer_itemspawn_markers.dm]: Ah, thanks for catching that. They're meant to sit there in processing objects yes.

File: code/modules/halo/weapons/forerunner/snipers.dm (hunk 2 above)
Comment (X0-11) [reply code/modules/halo/weapons/forerunner/snipers.dm:18]: The change is minor enough that I didn't even recall making it. I don't think it'll have a player facing effect.

File: code/modules/halo/vehicles/vehiclebase.dm (hunk 1 above)
Comment (X0-11) [reply code/modules/halo/vehicles/vehiclebase.dm]: Added.

**Review by bloxgate — APPROVED** [review]

File: baystation12.dme [hunk 4]
```diff
@@ -1758,7 +1758,6 @@
 #include "code\modules\halo\insurrection\x52_spawns.dm"
 #include "code\modules\halo\leader_support\job_overrides.dm"
 #include "code\modules\halo\leader_support\supply_drop.dm"
-#include "code\modules\halo\leader_support\supply_drop_evac.dm"
```
Comment (bloxgate) [inline baystation12.dme:1761]: Actually just noticed this. This file doesn't seem to be removed by this PR, but is unincluded. I just want to make sure that that is intentional.

Comment (X0-11) [reply baystation12.dme:1761]: Yeah that's unincluded on purpose. I'm pretty sure it's broken in every mode right now and i may fix it at a later date.

**Pull request merged**